accumulator/infra/main.tf

```diff
@@ -0,0 +1,300 @@
+}
+
+resource "aws_iam_instance_profile" "emr_profile" {
+  name = "emr_profile"
+  role = aws_iam_role.iam_emr_profile_role.name
+}
+
+data "aws_iam_policy_document" "iam_emr_service_policy" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:AuthorizeSecurityGroupEgress",
+      "ec2:AuthorizeSecurityGroupIngress",
+      "ec2:CancelSpotInstanceRequests",
+      "ec2:CreateNetworkInterface",
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateTags",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteTags",
+      "ec2:DescribeAvailabilityZones",
+      "ec2:DescribeAccountAttributes",
+      "ec2:DescribeDhcpOptions",
+      "ec2:DescribeInstanceStatus",
+      "ec2:DescribeInstances",
+      "ec2:DescribeKeyPairs",
+      "ec2:DescribeNetworkAcls",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DescribePrefixLists",
+      "ec2:DescribeRouteTables",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeSpotInstanceRequests",
+      "ec2:DescribeSpotPriceHistory",
+      "ec2:DescribeSubnets",
+      "ec2:DescribeVpcAttribute",
+      "ec2:DescribeVpcEndpoints",
+      "ec2:DescribeVpcEndpointServices",
+      "ec2:DescribeVpcs",
+      "ec2:DetachNetworkInterface",
+      "ec2:ModifyImageAttribute",
+      "ec2:ModifyInstanceAttribute",
+      "ec2:RequestSpotInstances",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RunInstances",
+      "ec2:TerminateInstances",
+      "ec2:DeleteVolume",
+      "ec2:DescribeVolumeStatus",
+      "ec2:DescribeVolumes",
+      "ec2:DetachVolume",
+      "iam:GetRole",
+      "iam:GetRolePolicy",
+      "iam:ListInstanceProfiles",
+      "iam:ListRolePolicies",
+      "iam:PassRole",
+      "s3:CreateBucket",
+      "s3:Get*",
+      "s3:List*",
+    ]
+
+  }
+}
+
+resource "aws_iam_role" "iam_emr_service_role" {
+  name               = "iam_emr_service_role"
+  assume_role_policy = data.aws_iam_policy_document.iam_emr_service_policy.json
+}
+
+# EMR
+######
+
+resource "aws_emr_cluster" "accumulator" {
+  name = "accumulator"
+  release_label = "emr-6.11.0"
+  applications = [ "Hadoop" ]
+  service_role = aws_iam_role.iam_emr_service_role.arn
+  ec2_attributes {
+    subnet_id                         = aws_subnet.private-1.id
+    emr_managed_master_security_group = aws_security_group.emr_master.id
+    emr_managed_slave_security_group  = aws_security_group.emr_core.id
+    instance_profile                  = aws_iam_instance_profile.emr_profile.arn
+  }
+
+  master_instance_group {
+    instance_type = "m5a.large"
+  }
+
+  core_instance_group {
+    instance_count = 1
+    instance_type  = "m5a.large"
+    # TODO: set up autoscaling policy w/ spot instances.
+    ebs_config {
+      size                 = "20"
+      type                 = "gp2"
+      volumes_per_instance = 1
+    }
+  }
+
+  bootstrap_action {
+    path = "s3://elasticmapreduce/bootstrap-actions/run-if"
```

Comment:
how did you you choose this bootstrap action script?

Reply:
This is a TODO, that's the EMR hello world bootstrap action.

We'll use this script to install the streaming JARS/NPM dependencies.

Reply:
* Unless we come up with a pre-bootstrapped custom AMI. That's usually preferable: it drastically decrease the boot time for the autoscaling workers.